src/main/java/seedu/address/logic/parser/AddTagCommandParser.java

```diff
@@ -0,0 +1,50 @@
+package seedu.address.logic.parser;
+
+import static seedu.address.logic.parser.CliSyntax.PREFIX_PRIORITY_STATUS;
+
+import java.util.stream.Stream;
+
+import seedu.address.commons.core.Messages;
+import seedu.address.commons.core.index.Index;
+import seedu.address.logic.commands.AddTagCommand;
+import seedu.address.logic.commands.Command;
+import seedu.address.logic.parser.exceptions.ParseException;
+import seedu.address.model.tag.PriorityTag;
+
+/**
+ * AddTagCommandParser parses the arguments and supplies these arguments
+ * to the AddTagCommand.
+ */
+public class AddTagCommandParser implements Parser<Command> {
+    @Override
+    public Command parse(String args) throws ParseException {
+        String[] indexWithTags = args.strip().split(" ", 2);
+        if (indexWithTags.length != 2) {
+            throw new ParseException(String.format(Messages.MESSAGE_INVALID_COMMAND_FORMAT,
+                    AddTagCommand.MESSAGE_USAGE));
+        }
+        try {
+            Index index = ParserUtil.parseIndex(indexWithTags[0]);
+            String tags = " " + indexWithTags[1];
+            ArgumentMultimap argMultiMap = ArgumentTokenizer.tokenize(tags, PREFIX_PRIORITY_STATUS);
+            if (!arePrefixesPresent(argMultiMap, PREFIX_PRIORITY_STATUS)
+                    || !argMultiMap.getPreamble().isEmpty()) {
+                throw new ParseException(String.format(Messages.MESSAGE_INVALID_COMMAND_FORMAT,
+                        AddTagCommand.MESSAGE_USAGE));
+            }
+            PriorityTag priorityTag = null;
+            String priorityStatus = argMultiMap.getValue(PREFIX_PRIORITY_STATUS).orElse(null);
+            if (priorityStatus != null) {
+                priorityTag = ParserUtil.parsePriorityTag(priorityStatus);
```

Comment:
In this case, if priority status does not exist, it will be null cos the priority tag has not been created yet. I cannot pass priority status into parsePriorityTag to handle it as it will throw nullpointerException.

Reply:
I agree with you but I think the reason why I made it as such is because I wanted to account for the future where there might be multiple prefixes. I think I can change it for now

Reply:
Does arePrefixesPresent take care of this already? Because if I'm not wrong, if user input does not have "p/" then arePrefixesPresent will return false, if p/ exists but nothing follows it, then arePrefixesPresent will return true and calling getValue().get() will return "" which can be passed into parsePriorityTag.